localstack-core/localstack/utils/analytics/usage.py

```diff
@@ -50,15 +50,47 @@ def aggregate(self) -> dict:
 
 class UsageCounter:
     """
-    Use this counter to count numeric values and perform aggregations
+    Use this counter to count numeric values
 
-    Available aggregations: min, max, sum, mean, median
+    Example:
+        my__counter = UsageCounter("lambda:somefeature")
+        my_counter.increment()
+        my_counter.increment()
+        my_counter.aggregate()  # returns {"count": 2}
+    """
+
+    state: int
+    namespace: str
+
+    def __init__(self, namespace: str):
+        self.enabled = not config.DISABLE_EVENTS
+        self.state = 0
+        self._counter = count(1)
+        self.namespace = namespace
+        collector_registry[namespace] = self
+
+    def increment(self):
+        # TODO: we should instead have different underlying datastructures to store the state, and have no-op operations
+        #  when config.DISABLE_EVENTS is set
+        if self.enabled:
+            self.state = next(self._counter)
+
+    def aggregate(self) -> dict:
+        # TODO: should we just keep `count`? "sum" might need to be kept for historical data?
+        return {"count": self.state, "sum": self.state}
+
+
+class TimingStats:
```

Comment:
Not a fan of the name TBH. Not everything collected here has to be a "timing". I like your suggestion of `NumericStats` better :+1: